src/openms/source/ANALYSIS/ID/SearchDatabase.cpp

```diff
@@ -0,0 +1,332 @@
+  
+  return all_frags;
+}
+
+SearchDatabase::SearchDatabase(const std::vector<FASTAFile::FASTAEntry>& entries) : DefaultParamHandler("SearchDatabase")
+{ 
+  vector<String> all_enzymes;  
+  ProteaseDB::getInstance()->getAllNames(all_enzymes);
+  vector<String> all_mods;
+  ModificationsDB::getInstance()->getAllSearchModifications(all_mods);
+  vector<string> tolerance_units{"Da", "ppm"};
+  defaults_.setValue("digestor_enzyme", "Trypsin", "Enzyme for digestion");
+  defaults_.setValidStrings("digestor_enzyme", ListUtils::create<std::string>(all_enzymes));
+  defaults_.setValue("missed_cleavages", 0, "missed cleavages for digestion");
+  defaults_.setValue("peptide_min_mass", 500, "minimal peptide mass for database");
+  defaults_.setValue("peptide_max_mass", 5000, "maximal peptide mass for database");
+  defaults_.setValue("peptide_min_length", 5, "minimal peptide length for database");
+  defaults_.setValue("peptide_max_length", 50, "maximal peptide length for database");
+  defaults_.setValue("fragment_min_mz", 150, "minimal fragment mz for database");
+  defaults_.setValue("fragment_max_mz", 2000, "maximal fragment mz for database");
+  defaults_.setValue("precursor_mz_tolerance", 2, "tolerance for precursor-MZ in search");
+  defaults_.setValue("fragment_mz_tolerance", 0.05, "tolerance for fragment-MZ in search");
+  defaults_.setValue("precursor_mz_tolerance_unit", "Da", "unit of tolerance for precursor-MZ");
+  defaults_.setValidStrings("precursor_mz_tolerance_unit", tolerance_units);
+  defaults_.setValue("fragment_mz_tolerance_unit", "Da", "unit of tolerance for fragment-MZ");
+  defaults_.setValidStrings("fragment_mz_tolerance_unit", tolerance_units);
+  defaults_.setValue("modifications_fixed", std::vector<std::string>{"Carbamidomethyl (C)"}, "Fixed modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)'");
+  defaults_.setValidStrings("modifications_fixed", ListUtils::create<std::string>(all_mods));
+  defaults_.setValue("modifications_variable", std::vector<std::string>{"Oxidation (M)"}, "Variable modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Oxidation (M)'");
+  defaults_.setValidStrings("modifications_variable", ListUtils::create<std::string>(all_mods));
+  defaults_.setValue("max_variable_mods_per_peptide", 2, "Maximum number of residues carrying a variable modification per candidate peptide");  
+
+  defaultsToParam_();
+
+  if (entries.size() == 0) return;
+
+  all_peptides_ = generate_peptides_(entries);
+
+  all_fragments_ = generate_fragments_();
+  
+  bucketsize_ = size_t(sqrt(all_fragments_.size())); //calculating bucketsize for balanced tree (performance)
+
+  for (size_t i = 0; i < all_fragments_.size(); i += bucketsize_)
+  {
+    bucket_frags_mz_.emplace_back(all_fragments_[i].fragment_mz_);
+  }
+  #pragma omp parallel
+  {
+    // sorting the fragments of each bucket by the precursor-MZ
+    #pragma omp for
```

Comment:
```suggestion
    // sorting the fragments of each bucket by the precursor-MZ
  #pragma omp parallel for
  {
```

Reply:
Is this { in line 200 needed. Cause our code only works without it?!

Reply:
or delete the corresponding `}` after the for loop (probably even better this way)